code/utils.py

```diff
@@ -20,12 +25,77 @@ def get_userid(u):
         return None
 
 
+def get_user_emails():
+    return user_id_lookup.keys()
+
+
 def populate_userid_mapping():
     """
-
+    populates the user_id_lookup table based on file employee_id_mapping.csv extracted from the db
     """
     mapping = np.loadtxt('../resources/employee_id_mapping.csv', dtype='str', delimiter=',', skiprows=1)
     for m in mapping:
         user_id_lookup[m[0]] = int(m[1])
 
 
+def plot_with_tsne(email_ids, embeddings, display_plot):
+    """
+    expects a list of email_ids and numpy ndarray of embeddings. The numpy ndarray should have shape L,D where D is the
+    size of embeddings and L is the number of users
+    :param email2embedding:
+    :return:
+    """
+    tsne = TSNE(verbose=1, method='exact')
+    start = time.time()
+    tsne_embs = tsne.fit_transform(embeddings)
+    end = time.time()
+    print('time taken by TSNE ', (end-start))
+
+    if display_plot:
+        fig, ax = plt.subplots()
+        scatter = ax.scatter(tsne_embs[:,0], tsne_embs[:,1], s=30)
+        tooltip = mpld3.plugins.PointLabelTooltip(scatter, labels=email_ids)
```

Comment:
A passing thought:

Let's change email_ids to labels?
And we could pass org hierarchy info along with email ids
Just need to think what all information we want to display.

Reply:
done